plugins/spi/SPIOutputTest.cpp

```diff
@@ -1139,3 +1143,275 @@ void SPIOutputTest::testCombinedAPA102ControlPixelBrightness() {
+                              };
+  OLA_ASSERT_DATA_EQUALS(EXPECTED3, arraysize(EXPECTED3), data, length);
+  OLA_ASSERT_EQ(3u, backend.Writes(0));
+
+  // test4
+  // tests what happens if fewer then needed color information are received
+  buffer.SetFromString("7, 9");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  // check that the returns are the same as test3 (nothing changed)
+  OLA_ASSERT_DATA_EQUALS(EXPECTED3, arraysize(EXPECTED3), data, length);
+  OLA_ASSERT_EQ(3u, backend.Writes(0));
+
+  // test5
+  // test with changed StartAddress
+  // set StartAddress
+  output.SetStartAddress(3);
+  // values 1 & 2 should not be visible in SPI data stream
+  buffer.SetFromString("1,2,3,4,5,6,7,8");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED5[] = { 0x92, 0x49, 0xA4,   // Pixel 1 Green (4)
+                                0x92, 0x49, 0x36,   // Pixel 1 Red (3)
+                                0x92, 0x49, 0xA6,   // Pixel 1 Blue (5)
+                                0x92, 0x49, 0xB6,   // Pixel 2 Green (7)
+                                0x92, 0x49, 0xB4,   // Pixel 2 Red (6)
+                                0x92, 0x4D, 0x24    // Pixel 2 Blue (8)
+                              };
+  OLA_ASSERT_DATA_EQUALS(EXPECTED5, arraysize(EXPECTED5), data, length);
+  OLA_ASSERT_EQ(4u, backend.Writes(0));
+  // change StartAddress back to default
+  output.SetStartAddress(1);
+
+  // test6
+  // Check nothing changed on the other output.
+  OLA_ASSERT_EQ(reinterpret_cast<const uint8_t*>(NULL),
+                backend.GetData(1, &length));
+  OLA_ASSERT_EQ(0u, backend.Writes(1));
+
+  // test7
+  // test for multiple ports
+  // StartFrame is only allowed on first port.
+  SPIOutput::Options option1(1, "second SPI Device");
+  // setup pixel_count to 2 (enough to test all cases)
+  option1.pixel_count = 2;
+  // setup SPIOutput
+  SPIOutput output1(m_uid, &backend, option1);
+  // set personality
+  output1.SetPersonality(this_test_personality);
+  // setup some 'DMX' data
+  buffer.SetFromString("1, 10, 100");
+  // simulate incoming data
+  output1.WriteDMX(buffer);
+  // get fake SPI data stream
+  data = backend.GetData(1, &length);
+  // this is the expected spi data stream:
+  // StartFrame is missing --> port is >0 !
+  const uint8_t EXPECTED7[] = { 0x92, 0x4D, 0x34,   // Pixel 1 Green (10)
+                                0x92, 0x49, 0x26,   // Pixel 1 Red (1)
+                                0x9B, 0x49, 0xA4,   // Pixel 1 Blue (100)
+                                0x92, 0x49, 0x24,   // Pixel 2 Green (0)
+                                0x92, 0x49, 0x24,   // Pixel 2 Red (0)
+                                0x92, 0x49, 0x24    // Pixel 2 Blue (0)
+                              };
+  // check for Equality
+  OLA_ASSERT_DATA_EQUALS(EXPECTED7, arraysize(EXPECTED7), data, length);
+  // check if the output writes are 1
+  OLA_ASSERT_EQ(1u, backend.Writes(1));
+}
```

Comment:
This protocol doesn't seem to have a start frame, so I'm not sure this test is required.